### PR TITLE
Issue #1054 | Added warning to PollingConditions documentation as wel…

### PR DESCRIPTION
**Pull request by macieg**
…l as a test that explains the issue

---
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/1055)

### CODE REVIEW

**Comment by szpak** [comment]
Thanks for your PR. I've just slightly reworked it with `@PendingFeature` to fail once it is implemented.

File: spock-core/src/main/java/spock/util/concurrent/PollingConditions.java [hunk 1]
```diff
@@ -40,6 +40,12 @@
  *   assert machine.efficiency >= 0.9
  * }
  * </pre>
+ *
+ * Warning! Avoiding assert keyword in the clojure is only possible if the conditions object type is known
```
Comment (Vampire) [inline spock-core/src/main/java/spock/util/concurrent/PollingConditions.java]: It is "closure", not "clojure".
Clojure is a Lisp dialect.

Comment (szpak) [reply spock-core/src/main/java/spock/util/concurrent/PollingConditions.java]: My bad, I missed it.

Comment (macieg) [reply spock-core/src/main/java/spock/util/concurrent/PollingConditions.java]: That's true, sorry!

**Pull request merged**